failure to build on centos

these-1.1.1 is failing to compile (as a dependency) on CentOS images using ghc-8.8.3 (and otherwise using a freeze file that matches stackage 16.5) with

```
Building library for these-1.1.1..

src/Data/Functor/These.hs:13:0: error:
     error: missing binary operator before token "("
     #elif MIN_VERSION_transformers(0,5,0)
     ^
   |
13 | #elif MIN_VERSION_transformers(0,5,0)
   | ^
```

which might possibly be related to the tooling available on that platform.

The only thing I can think of is maybe the empty newline on the previous line is rejected, somehow :man_shrugging:

I'll look into this. I have a script to run cabal inside a docker image of any other OS that you might be interested in. Yes, a simple 50 line bash script can replace an entire `stack` feature! Who would have guessed?

https://gitlab.com/fommil/dotfiles/-/blob/master/bin/cabal-docker

FYI this is the more detailed error

```
src/Data/Functor/These.hs:13:0: error:
     error: missing binary operator before token "("
     #elif MIN_VERSION_transformers(0,5,0)
     ^
   |
13 | #elif MIN_VERSION_transformers(0,5,0)
   | ^

src/Data/Functor/These.hs:16:0: error:
     error: missing binary operator before token "("
     #elif MIN_VERSION_transformers_compat(0,5,0) && !MIN_VERSION_transformers(0,4,0)
     ^
   |
16 | #elif MIN_VERSION_transformers_compat(0,5,0) && !MIN_VERSION_transformers(0,4,0)
   | ^

src/Data/These.hs:11:0: error:
     error: missing binary operator before token "("
     #elif MIN_VERSION_transformers(0,5,0)
     ^
   |
11 | #elif MIN_VERSION_transformers(0,5,0)
   | ^

src/Data/These.hs:14:0: error:
     error: missing binary operator before token "("
     #elif MIN_VERSION_transformers_compat(0,5,0) && !MIN_VERSION_transformers(0,4,0)
     ^
   |
14 | #elif MIN_VERSION_transformers_compat(0,5,0) && !MIN_VERSION_transformers(0,4,0)
   | ^

src/Data/Functor/These.hs:1:1: error:
    `gcc' failed in phase `C pre-processor'. (Exit code: 1)
  |
1 | {-# LANGUAGE CPP                #-}
  | ^

src/Data/These.hs:1:1: error:
    `gcc' failed in phase `C pre-processor'. (Exit code: 1)
  |
1 | {-# LANGUAGE CPP                #-}
  | ^
```

--- a/these/src/Data/Functor/These.hs
+++ b/these/src/Data/Functor/These.hs
@@ -9,12 +9,14 @@
 
 #if MIN_VERSION_base(4,9,0)
 #define LIFTED_FUNCTOR_CLASSES 1
-
-#elif MIN_VERSION_transformers(0,5,0)
+#else
+#if MIN_VERSION_transformers (0,5,0)
 #define LIFTED_FUNCTOR_CLASSES 1
-
-#elif MIN_VERSION_transformers_compat(0,5,0) && !MIN_VERSION_transformers(0,4,0)
+#else
+#if MIN_VERSION_transformers_compat(0,5,0) && !MIN_VERSION_transformers(0,4,0)
 #define LIFTED_FUNCTOR_CLASSES 1
+#endif
+#endif
 #endif
 module Data.Functor.These (
     These1 (..),

--- a/these/src/Data/These.hs
+++ b/these/src/Data/These.hs
@@ -7,13 +7,16 @@
 
 #if MIN_VERSION_base(4,9,0)
 #define LIFTED_FUNCTOR_CLASSES 1
-
-#elif MIN_VERSION_transformers(0,5,0)
+#else
+#if MIN_VERSION_transformers(0,5,0)
 #define LIFTED_FUNCTOR_CLASSES 1
-
-#elif MIN_VERSION_transformers_compat(0,5,0) && !MIN_VERSION_transformers(0,4,0)
+#else
+#if MIN_VERSION_transformers_compat(0,5,0) && !MIN_VERSION_transformers(0,4,0)
 #define LIFTED_FUNCTOR_CLASSES 1
 #endif
+#endif
+#endif
+
 module Data.These (
       These(..)
 

